src/sentry/tasks/integrations/github/language_parsers.py

```diff
@@ -216,6 +216,118 @@ def generate_function_name_conditions(function_names: list[str], stack_frame: in
+        anonymous_function_regex = r"^@@.*@@.*\s+\$(?P<fnc>\w+)\s*=\s*function\s*\(.*$"
+        arrow_function_regex = r"^@@.*@@.*\s+\$(?P<fnc>\w+)\s*=\s*fn\([^)]*\)\s*=>.*$"
+
+        regexes = [
+            function_declaration_regex,
+            arrow_function_regex,
+            anonymous_function_regex,
+        ]
+
+        functions = set()
+        for regex in regexes:
+            functions.update(set(re.findall(regex, patch, flags=re.M)))
+
+        return functions
+
+    @staticmethod
+    def _get_function_name_conditions(stackframe_level: int, function_names: list[str]):
+        """
+        For PHP we need a special case of matching both for the function name itself and for
+        "::" + the function name, because sometimes Snuba stores the function name as "namespace::FunctionName".
+        """
+        prepended_function_names = ["%::" + function_name for function_name in function_names]
+        function_name_conditions = [
+            Condition(
+                stackframe_function_name(stackframe_level),
+                Op.LIKE,
+                function_name,
+            )
+            for function_name in prepended_function_names
+        ]
+        function_name_conditions.append(
+            Condition(
+                stackframe_function_name(stackframe_level),
+                Op.IN,
+                function_names,
+            ),
+        )
+        return function_name_conditions
+
+    @staticmethod
+    def _get_function_name_functions(stackframe_level: int, function_names: list[str]):
+        """
+        This is used in the multi_if. We need to account for the special Javascript cases in order to
+        properly fetch the function name -- "namespace::FunctionName" or simply "functionName" depending
+        on what matches in the stack trace.
+        """
+        prepended_function_names = ["%::" + function_name for function_name in function_names]
+        function_name_conditions = [
+            Function(
+                "like",
+                [
+                    stackframe_function_name(stackframe_level),
+                    function_name,
+                ],
+            )
+            for function_name in prepended_function_names
+        ]
+        function_name_conditions.append(
+            Function(
+                "in",
+                [
+                    stackframe_function_name(stackframe_level),
+                    function_names,
+                ],
+            )
+        )
+        return function_name_conditions
+
+    @staticmethod
+    def generate_multi_if(function_names: list[str]) -> list[Function]:
+        """
+        Fetch the function name from the stackframe that matches a name within the list of function names.
+        """
+        multi_if = []
+        for i in range(-STACKFRAME_COUNT, 0):
+            # if, then conditions
+            stackframe_function_name_conditions = PHPParser._get_function_name_functions(
```

Comment:
i see, if we have it as `@classmethod` you could do `self._get_function_name_functions` here